src/rust/engine/src/scheduler.rs

```diff
@@ -331,6 +336,23 @@ impl Scheduler {
     })
   }
 
+  async fn wait_for_tail_tasks(tasks: Vec<BoxFuture<'static, ()>>) {
+    if !tasks.is_empty() {
+      log::trace!("waiting for {} tail tasks to complete", tasks.len());
+
+      let joined_tail_tasks_fut = futures::future::join_all(tasks);
+      let timeout_fut = time::timeout(Duration::from_secs(5), joined_tail_tasks_fut);
```

Comment:
This should probably be configurable.

Reply:
Probably. Maybe `--tail-tasks-timeout`. It's hard to have a less vague name because tail tasks will be used for both streaming workunit handlers & remote cache writes.

Reply:
--session-end-task-timeout ?

Reply:
Something like that, yea. Or, if this ends up as an explicit method on `Executor`, it's possible that a tail task should specify its own timeout at that point: then you could make the setting specific to cache writes: `--remote-cache-write-tail-timeout`, etc.

Reply:
I went with `--session-end-tasks-timeout` (plural) for now.